src/main/java/jp/co/soramitsu/sora/json/Flattener.java

```diff
@@ -0,0 +1,154 @@
+package jp.co.soramitsu.sora.json;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.databind.node.ObjectNode;
+import java.util.stream.IntStream;
+import lombok.val;
+
+
+public class Flattener {
+
+  private char dictkey = '/';
+  private char arrkey = '_';
+  private ObjectMapper mapper = new ObjectMapper();
+
+  private void flatten(JsonNode root, String path, ObjectNode out) {
+    if (root.isObject()) {
+      if (root.size() == 0) {
+        // it is empty object
+        out.set(path, root);
+        return;
+      }
+
+      val fields = root.fields();
+      fields.forEachRemaining(
+          field -> flatten(field.getValue(), path + dictkey + sanitize(field.getKey()), out));
+    } else if (root.isArray()) {
+      if (root.size() == 0) {
+        // it is empty array
+        out.set(path, root);
+        return;
+      }
+
+      IntStream
+          .range(0, root.size())
+          .forEachOrdered(i -> flatten(root.get(i), path + arrkey + i, out));
+    } else if (root.isValueNode()) {
+      out.set(path, root);
+
+    } else {
+      throw new FlattenerException("type is neigher object, array or value");
+    }
+  }
+
+  private void deflatten(JsonNode root, ObjectNode out) {
+    throw new RuntimeException("not implemented");
+  }
+
+  public Flattener() {
+  }
+
+  public Flattener(ObjectMapper mapper) {
+    this.mapper = mapper;
+  }
+
+  public Flattener setObjectMapper(ObjectMapper mapper) {
```

Comment:
please use lombok annotations for setters

Reply:
This is not just a setter, but kind of builder method, so you can write 

```
new Flattener().setObjectMapper(new ObjectMapper()).setDictionaryKey('`').flatten(...)
```